src/main/java/com/epam/ta/reportportal/core/analyzer/auto/impl/SearchLogServiceImpl.java

```diff
@@ -129,8 +130,10 @@ public SearchLogServiceImpl(ProjectRepository projectRepository, LaunchRepositor
 				value.getLogs().add(TO_LOG_ENTRY.apply(log));
 				return value;
 			});
-			Launch launch = launchRepository.findById(log.getTestItem().getLaunchId())
-					.orElseThrow(() -> new ReportPortalException(ErrorType.LAUNCH_NOT_FOUND, log.getTestItem().getLaunchId()));
+			Long launchId = ofNullable(log.getTestItem()
```

Comment:
why is it better than the previous variant?

Reply:
Retrise doesn't have launch Id and not supported with search log functionality. So we check it here